libc/test/src/time/gettimeofday_test.cpp

```diff
@@ -15,24 +15,31 @@
 
 namespace cpp = LIBC_NAMESPACE::cpp;
 
+using LIBC_NAMESPACE::testing::tlog;
+
 TEST(LlvmLibcGettimeofday, SmokeTest) {
   using LIBC_NAMESPACE::testing::ErrnoSetterMatcher::Succeeds;
   void *tz = nullptr;
-  timeval tv;
 
   suseconds_t sleep_times[2] = {200, 1000};
```

Comment:
Since these are magic values anyway, I don't think it makes sense to double them further down.  Perhaps we should just pick 1000, 2000 here and comment above that sub-second clocks seem to get confused.

I remember awhile ago there were some motherboards whose TSCs would run backwards for some strange reason.  I haven't been able to find any indication that this is still a problem, which makes me wonder if there's a different bug here as well.  I don't know enough about system clocks, though.

Reply:
So far looks like the failures are on `sleep_time = 1000`.  Do you think it is because of the first sleep by 200us ?

Reply:
> I remember awhile ago there were some motherboards whose TSCs would run backwards for some strange reason. I haven't been able to find any indication that this is still a problem, which makes me wonder if there's a different bug here as well. I don't know enough about system clocks, though.

Are you talking about 1. `TSC` (clock internal to the CPU core and read via instruction such as `rdtsc`) or 2. the BIOS `RTC` (clock that runs on battery on the motherboard)?

Possible explanations depending on the answer:
1. the counter wrapped around (hence looking like stepping back) or the process has been migrated to another core (with a different counter value).
2. it can come from background NTP synchronization that resets the clock to a different point in time.

Reply:
So maybe the solution is if the time diff is negative, we can simply try again?

Reply:
SGTM it's unlikely that it would step back in time twice in a row for the reasons I invoked. If so, something's fishy for sure. `gettimeofday` is probably based on `RTC` so it's probably `NTP`.

Reply:
I think maybe we're going about this wrong.  It's not our job to test the kernel for accuracy.  Probably a single call to make sure that it returns a number is fine (that way we know that our call to the sysall at least works).  If syscalls are implemented like https://xkcd.com/221/ then that's not our problem (and we wouldn't want to check that on every environment in which this might be implemented anyway).

WDYT?

Reply:
> It's not our job to test the kernel for accuracy
Yeah indeed
> Probably a single call to make sure that it returns a number is fine
👍

Reply:
SGTM